webapp/channels/src/components/interactive_dialog/interactive_dialog.tsx

```diff
@@ -1,53 +1,61 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
+import type {DialogElement as TDialogElement} from '@mattermost/types/integrations';
+
 import {
     checkDialogElementForError,
     checkIfErrorsMatchElements,
 } from 'mattermost-redux/utils/integration_utils';
 
 import SpinnerButton from 'components/spinner_button';
 
+import type EmojiMap from 'utils/emoji_map';
 import {localizeMessage} from 'utils/utils';
 
 import DialogElement from './dialog_element';
 import DialogIntroductionText from './dialog_introduction_text';
 
-export default class InteractiveDialog extends React.PureComponent {
-    static propTypes = {
-        url: PropTypes.string.isRequired,
-        callbackId: PropTypes.string,
-        elements: PropTypes.arrayOf(PropTypes.object),
-        title: PropTypes.string.isRequired,
-        introductionText: PropTypes.string,
-        iconUrl: PropTypes.string,
-        submitLabel: PropTypes.string,
-        notifyOnCancel: PropTypes.bool,
-        state: PropTypes.string,
-        onExited: PropTypes.func,
-        actions: PropTypes.shape({
-            submitInteractiveDialog: PropTypes.func.isRequired,
-        }).isRequired,
-        emojiMap: PropTypes.object.isRequired,
+export type Props = {
+    url: string;
+    callbackId?: string;
+    elements?: TDialogElement[];
+    title: string;
+    introductionText?: string;
+    iconUrl?: string;
+    submitLabel?: string;
+    notifyOnCancel?: boolean;
+    state?: string;
+    onExited?: () => void;
+    actions: {
+        submitInteractiveDialog: any;
```

Comment:
type of `submitInteractiveDialog:` can be obtained its an action defined in webapp/channels/src/packages/mattermost-redux/src/actions/integrations.ts

Reply:
I tried that! it was giving error ! thats why changed it to any !
![image](https://github.com/mattermost/mattermost/assets/93595710/05033f63-965f-4f6b-849d-377a01b12211)

Reply:
ok i have resolved it ! have a look at it !